### ISSUE TITLE
request.path starts with "//" after updating to v10

### ISSUE THREAD

**Issue by MathieuDerelle** [airbrake/airbrake#1072]
I've updated airbrake to avoid deprecation warnings
https://github.com/airbrake/airbrake-ruby/issues/532

when opening a page in active admin, `request.path` is now
`//admin/users/1770929`
before it was
`/admin/users/1770929`

The only change in the gemfile lock is concerning airbrake

before
```
    airbrake (9.3.0)
      airbrake-ruby (~> 4.5)
    airbrake-ruby (4.13.0)
```

after
```
    airbrake (10.0.1)
      airbrake-ruby (~> 4.13)
    airbrake-ruby (4.13.1)
```

edit : a bit more details
- its a rails 3.2.22.5 app with an engine "backoffice"
- it happens regardless how many starting slashes I enter in the URL in my browser

**Comment by kyrylo** [comment]
Sorry for the late reply, this fell off my radar.

Do you have a reproducible case? I've never used active admin and I currently don't fully understand this report. Do you mean Airbrake somehow messes up Rack request and appends an extra slash to `request.path`? Are we talking about `Rack::Request`?

**Comment by MathieuDerelle** [comment]
`request` is rails' `ActionDispatch::Request` (which inherits from `Rack::Request` yeah)

I'm not quite sure what going on, when logging `request`, it seems fine
`` 
```
#<ActionDispatch::Request ... "PATH_INFO"=>"/admin/pdfs/811"
...
@filtered_parameters={"action"=>"show", "controller"=>"admin/pdfs", "id"=>"811"}, @method="GET", @fullpath="/admin/pdfs/811", @port=15800, @protocol="http://", @symbolized_path_params={:action=>"show", :controller=>"admin/pdfs", :id=>"811"}>
```

but logging `request.path` just after that gives `"//admin/pdfs/811"`

looking at `path` definition
https://github.com/rack/rack/blob/1f5763de6a9fe515ff84992b343d63c88104654c/lib/rack/request.rb#L511-L513
I can confirm that `@env["SCRIPT_NAME"]` is now equal to `'/'`

**Comment by MathieuDerelle** [comment]
:o 

juste noticed if was prolly fixed with the other issue

**Comment by MathieuDerelle** [comment]
just tried 10.0.4 and it did not fix this issue

**Comment by MathieuDerelle** [comment]
any insight on what could modify `@env["SCRIPT_NAME"]` in your code ?

I'll try to found out which commit exactly is in fault but since there's a gap in major, im not sure if the commits are "continuous" from 9 to 10

**Comment by kyrylo** [comment]
What do you mean by "continuous" and "gap"?

We don't modify `@env["SCRIPT_NAME"]` directly.

Unfortunately, I don't know the reason of this. I would really appreciate if you could find the commit that breaks it for you. We've had a similar issue in the past: https://github.com/airbrake/airbrake/issues/1091. Perhaps, that's what you were referring to.

**Comment by MathieuDerelle** [comment]
When coding v10, did you start from scratch ? or is it the same base code as v9 ?
That's what I mean with "continuous" and "gap"

**Comment by kyrylo** [comment]
No, v10 is a natural continuation of v9. It has breaking changes but it's still the same old v9.

**Comment by MathieuDerelle** [comment]
after investigation, the problem appeared with this commit 8bf9934f2627745a7d1fd90135219c1838f5e2e0

a couple other things
- the issue is not related to active admin : I originally found out the issue on our "backoffice" engine (which uses active admin) but the issue is also present on the other engines (which do not use active admin)
- the issue is also present in controllers without engine

**Comment by MathieuDerelle** [comment]
> We've had a similar issue in the past: #1091. Perhaps, that's what you were referring to.

You're correct, that was the one I was speaking of with

> juste noticed it was prolly fixed with the other issue

Sorry to not link it

**Comment by kyrylo** [comment]
@MathieuDerelle is it possible for you to share your `config/routes.rb`? You can email it to me if it's confidential: `kyrylo@airbrake.io`.

**Comment by kyrylo** [comment]
Gentle ping @MathieuDerelle

**Comment by MathieuDerelle** [comment]
sorry for the late answer

I spent several hours trying to create a reproduction repo but could not do it with a new rails project.
So I started with the faulty repo and dropped almost everything from it.

[more details on how to reproduce in the README](https://github.com/MathieuDerelle/issue-example-for-airbrake-1)

N.B. : I could not make it work without a DB, so you'll need one (it does not matter if there are no tables)
